ninja-core/src/main/java/ninja/template/directives/TemplateEngineFreemarkerAuthenticityTokenDirective.java

```diff
@@ -0,0 +1,40 @@
+package ninja.template.directives;
+
+import java.io.IOException;
+import java.io.Writer;
+import java.util.Map;
+
+import ninja.Context;
+import freemarker.core.Environment;
+import freemarker.template.TemplateDirectiveBody;
+import freemarker.template.TemplateDirectiveModel;
+import freemarker.template.TemplateException;
+import freemarker.template.TemplateModel;
+
+/**
+ * 
+ * @author svenkubiak
+ *
+ */
+@SuppressWarnings("rawtypes")
+public class TemplateEngineFreemarkerAuthenticityTokenDirective implements TemplateDirectiveModel {
+    private String authenticityToken;
+
+    public TemplateEngineFreemarkerAuthenticityTokenDirective(Context context) {
+        this.authenticityToken = context.getSession().getAuthenticityToken();
```

Comment:
please do this in the execute method. Otherwise you are always adding the __AT to the session - even if it is not needed...